File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/db/dao/LectureDao.java
package com.vladislavmyasnikov.courseproject.data.db.dao;

import com.vladislavmyasnikov.courseproject.data.db.entity.LectureEntity;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface LectureDao {

    @Query("SELECT * FROM lectures")
    LiveData<List<LectureEntity>> loadLectures();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertLectures(List<LectureEntity> lectures);
}

File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/db/dao/LectureDao.kt
package com.vladislavmyasnikov.courseproject.data.db.dao

import androidx.room.*
import com.vladislavmyasnikov.courseproject.data.db.entities.LectureEntity
import com.vladislavmyasnikov.courseproject.data.db.entities.LectureWithTasksEntity

@Dao
interface LectureDao {

    @Query("SELECT * FROM lectures")
    fun loadLectures(): List<LectureEntity>

    @Query("SELECT * FROM lectures")
    fun loadLecturesWithTasks(): List<LectureWithTasksEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertLectures(lectures: List<LectureEntity>)
}
